src/views/Generator/Generator.tsx

```diff
@@ -14,44 +14,38 @@ import { TestRuleContainer } from './TestRuleContainer'
 import {
   useGeneratorParams,
   useIsGeneratorDirty,
-  useLoadGeneratorFile,
-  useLoadHarFile,
+  useGeneratorFile,
   useSaveGeneratorFile,
 } from './Generator.hooks'
 import { GeneratorControls } from './GeneratorControls'
-import { useToast } from '@/store/ui/useToast'
-import { getRoutePath } from '@/routeMap'
 import { UnsavedChangesDialog } from './UnsavedChangesDialog'
 import { RuleEditor } from './RuleEditor'
 import { getFileNameWithoutExtension } from '@/utils/file'
-import log from 'electron-log/renderer'
+import { useToast } from '@/store/ui/useToast'
+import { getRoutePath } from '@/routeMap'
 
 export function Generator() {
   const selectedRule = useGeneratorStore(selectSelectedRule)
-
-  const setGeneratorFile = useGeneratorStore((store) => store.setGeneratorFile)
-
+  const setGenerator = useGeneratorStore((store) => store.setGenerator)
   const showToast = useToast()
   const navigate = useNavigate()
 
   const { fileName } = useGeneratorParams()
 
-  const {
-    data: generatorFileData,
-    isLoading: isLoadingGenerator,
-    error: generatorError,
-  } = useLoadGeneratorFile(fileName)
-
-  const {
-    data: recording,
-    isLoading: isLoadingRecording,
-    error: harError,
-  } = useLoadHarFile(generatorFileData?.recordingPath)
+  const { isLoading } = useGeneratorFile({
+    fileName,
+    onSuccess: setGenerator,
+    onError: () => {
+      showToast({
+        title: 'Failed to load generator',
+        status: 'error',
+      })
+      navigate(getRoutePath('home'), { replace: true })
+    },
+  })
```

Comment:
Not that I'm against this approach, I actually liked `onSuccess` and `onError` callbacks, but if we decide to follow this pattern with other hooks it would add duplication of setting up those callbacks or require custom wrappers for tanstack hooks. 

Have you considered using tanstack's proposed [migration guide](https://github.com/TanStack/query/discussions/5279), to use `useEffect` hooks instead?

```suggestion
  const { isLoading, data, isSuccess, isError } = useGeneratorFile({ fileName })

  useEffect(() => {
    if (isSuccess && data) {
      setGenerator(data)
    }
  }, [isSuccess, data, setGenerator])

  useEffect(() => {
    if (isError) {
      showToast({
        title: 'Failed to load generator',
        status: 'error',
      })
      navigate(getRoutePath('home'), { replace: true })
    }
  }, [isError, navigate, showToast])
```